dask/core.py

```diff
@@ -41,6 +41,81 @@ def istask(x):
     return isinstance(x, tuple) and x and callable(x[0])
 
 
+def isdag(d, keys):
+    # Stack-based depth-first search traversal
+    if not isinstance(keys, list):
+        keys = [keys]
+    completed = builtins.set()
+    seen = builtins.set()
+    for key in keys:
+        if key in completed:
+            continue
+        nodes = [key]
+        while nodes:
+            cur = nodes[-1]
+            if cur in completed:
+                nodes.pop()
+                continue
+            seen.add(cur)
+
+            next_nodes = []
+            for nxt in get_dependencies(d, cur):
+                if nxt not in completed:
+                    if nxt in seen:
+                        return False
+                    next_nodes.append(nxt)
+
+            if next_nodes:
+                nodes.extend(next_nodes)
+            else:
+                completed.add(cur)
+                nodes.pop()
+        return True
+
+
+def _get_task(d, task, maxdepth=1000):
+    # non-recursive.  DAG property is checked upon reaching maxdepth.
+    depth = 0
+    _iter = lambda *args: iter(args)
+    stack = (task[0], list(task[:0:-1]), [], None)
```

Comment:
Note that `task` may be a nested list.  We should emit results in the same configuration (e.g. possibly nested list of results).